integration_test/validate_metadata_test.go

```diff
@@ -61,18 +69,19 @@ func TestRequireMetadataForAllThirdPartyApps(t *testing.T) {
 }
 
 func TestThirdPartyPublicUrls(t *testing.T) {
-	err := walkThirdPartyApps(func(contents []byte) error {
-		integrationMetadata := &metadata.IntegrationMetadata{}
-		err := metadata.UnmarshalAndValidate(contents, integrationMetadata)
-		if integrationMetadata.PublicUrl == "" {
-			// The public doc isn't available yet.
-			return nil
-		}
-		if err != nil {
-			return err
-		}
-		t.Run(integrationMetadata.ShortName, func(t *testing.T) {
+	err := walkThirdPartyApps(func(fullPath string, contents []byte) error {
+		app := path.Base(path.Dir(fullPath))
+		t.Run(app, func(t *testing.T) {
 			t.Parallel()
+			integrationMetadata := &metadata.IntegrationMetadata{}
+			err := metadata.UnmarshalAndValidate(fullPath, contents, integrationMetadata)
+			if integrationMetadata.PublicUrl == "" {
+				// The public doc isn't available yet.
+				return
+			}
+			if err != nil {
+				t.Error(err)
+			}
 			r, err := http.Get(integrationMetadata.PublicUrl)
```

Comment:
We had a recent flake here with 5xx errors failing the test. Is this a good opportunity to either add retries or to move it to an integration test?

Reply:
As far as I can see, adding retries properly is a can of worms (what conditions should we retry? immediately or with backoff? how many times/how long? implement your own on top of `net/http` or add a dependency on a more sophisticated library like https://github.com/hashicorp/go-retryablehttp?).  Let's do that in a separate PR?